app/questionnaire/path_finder.py

```diff
@@ -91,19 +94,50 @@ def _remove_rule_answers(self, goto_rule):
                     self.answer_store.remove(answer_id=condition['id'],
                                              answer_instance=0,)
 
-    def get_routing_path(self, group_id=None, group_instance=0):
+    def get_routing_path(self, group_id, group_instance=0):
         """
         Returns a list of the block ids visited based on answers provided
         :return: List of block location dicts
         """
-        if group_id is None:
-            group_id = SchemaHelper.get_first_group_id(self.survey_json)
+        self.get_full_routing_path()
+
+        routing_path = None
+
+        for i, location in enumerate(self.full_routing_path):
+            if location.group_id == group_id and location.group_instance == group_instance:
+                routing_path = self.full_routing_path[i:]
+                break
+
+        if routing_path:
+            return routing_path
+
```

Comment:
use `else:` for the rest?

Reply:
Im not a fan of an `else` it generally indicates that the method does too much

Reply:
😜 may I suggest this method might be doing too much then

Reply:
not any more.

Reply:
👍